articles/quickstart/native/xamarin-ios/01-login.md

```diff
@@ -0,0 +1,171 @@
+
+```text
+YOUR_BUNDLE_IDENTIFIER://${account.namespace}/ios/YOUR_BUNDLE_IDENTIFIER/callback
+```
+
+Replace `YOUR_BUNDLE_IDENTIFIER` in the code sample above with the actual Bundle Identifier for your application, e.g. `com.mycompany.myapplication`.
+
+Finally, you will need to handle the callback to your application in your `AppDelegate` by calling the `Send` method of the `ActivityMediator` singleton, passing along the url sent in. This will allow the Auth0 OIDC Client library to complete the authentication process:
+
+```csharp
+using Auth0.OidcClient;
+
+[Register("AppDelegate")]
+public class AppDelegate : UIApplicationDelegate
+{
+	public override bool OpenUrl(UIApplication application, NSUrl url, string sourceApplication, NSObject annotation)
+	{
+		ActivityMediator.Instance.Send(url.AbsoluteString);
+
+		return true;
+	}
+}
+```
+
+## Integration
+
+To integrate Auth0 login into your application, simply instantiate an instance of the `Auth0Client` class, configuring the Auth0 Domain and Client ID, and also passing the instance uf your `UIViewController` as the `Controller` property: 
+
+${snippet(meta.snippets.setup)}
+
+You can then call the `LoginAsync` method to log the user in:
+
+${snippet(meta.snippets.use)}
+
+<div class="phone-mockup"><img src="/media/articles/native-platforms/xamarin-ios/lock-widget-screenshot.png" alt="Lock UI"></div>
+
+## Accessing the User's Information
+
+The returned login result will indicate whether authentication was successful, and if so contain the tokens and claims of the user.
+
+### Authentication Error
+
+You can check the `IsError` property of the result to see whether the login has failed. The `ErrorMessage` will contain more information regarding the error which occurred.
+
+```csharp
+var loginResult = await client.LoginAsync();
+
+if (loginResult.IsError)
+{
+    Debug.WriteLine($"An error occurred during login: {loginResult.Error}")
+}
+```
+
+### Accessing the tokens
+
+On successful login, the login result will contain the `id_token` and `access_token` in the `IdentityToken` and `AccessToken` properties respectively.
+
+```csharp
+var loginResult = await client.LoginAsync();
+
+if (!loginResult.IsError)
+{
+    Debug.WriteLine($"id_token: {loginResult.IdentityToken}");
+    Debug.WriteLine($"access_token: {loginResult.AccessToken}");
+}
+```
+
+### Obtaining the User Information
+
+On successful login, the login result will contain the user information in the `User` property, which is a [ClaimsPrincipal](https://msdn.microsoft.com/en-us/library/system.security.claims.claimsprincipal(v=vs.110).aspx).
+
+To obtain information about the user, you can query the claims. You can for example obtain the user's name and email address from the `name` and `email` claims:
+
+```csharp
+if (!loginResult.IsError)
+{
+    Debug.WriteLine($"name: {loginResult.User.FindFirst(c => c.Type == "name")?.Value}");
```

Comment:
Once again, I am returning the Login result exactly as-is from the underlying library. The `User` object is `ClaimsPrincipal` https://msdn.microsoft.com/en-us/library/system.security.claims.claimsprincipal(v=vs.110) which is a standard .NET class.

Returning this and letting the user access the claims they need is consistent with the other .NET platforms such as ASP.NET